canon-capt: init at 0.1.4.2-GxB

cryptoluks opened the pull request:
Add canon-capt driver package. This is a community-driven driver for Canon CAPT-based printers, based on reverse engineering attempts. Note that this is not the official Canon driver.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RaghavSood commented on pkgs/by-name/ca/canon-capt/package.nix:
```diff
@@ -0,0 +1,68 @@
+  pname = "canon-capt";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/mounaiban/captdriver/archive/8ecc3cde1ae9a20dcb015994bb0fea0dbefa2b83.zip";
+    sha256 = "sha256-QE9GiRC5ExH+LlooJ2eEFtsmA1aWzRoRfxh92EX6Tb8=";
+  };
```
You should use fetchFromGitHub - if you can't, maybe add a comment on why it won't work for this package.

RaghavSood commented on pkgs/by-name/ca/canon-capt/package.nix:
```diff
@@ -0,0 +1,68 @@
+  # Source: https://github.com/NixOS/nixpkgs/blob/9997402000a82eda4327fde36291234118c7515e/pkgs/misc/drivers/hplip/default.nix#L160
+  CUPS_DATADIR = "${cups}/share/cups";
+
+  unpackPhase = ''
+    runHook preUnpack
+    unzip $src
+    mv captdriver-*/* .
+    runHook postUnpack
+  '';
```
You can probably skip all of this if you use fetchFromGitHub, it unpacks the source for you

RaghavSood reviewed (commented):
The derivation is also missing a `meta` section - please add that

marcusramberg reviewed (commented):
Thanks for your contribution. Don't have a canon printer so can't actually test this, but it builds on x86-linux at least. Please add a meta section.

marcusramberg commented:
Sorry, I see my comment was a bit behind the times. Anyways, I think adding to the maintainer list should be done in a separate commit before the init commit.

cryptoluks committed:
maintainers: add cryptoluks

cryptoluks committed:
canon-capt: init at 0.1.4.2-GxB

Add canon-capt driver package. This is a community-driven
driver for Canon CAPT-based printers, based on reverse engineering
attempts. Note that this is not the official Canon driver.
